third_party/llvm-project/readme.txt

```diff
@@ -0,0 +1,19 @@
+This folder contains files from LLVM. See LICENSE.TXT for their license.
+
+These files were synced from
+
+commit 6c86d6efaf129c42d37121f1e7e9a7adffb54c1a (origin/master, master)
+Author: Craig Topper <craig.topper@intel.com>
+Date:   Mon Nov 11 16:30:42 2019 -0800
+
+    [X86] Remove some else branches after checking for !useSoftFloat() that set operations to Expand.
+    
+    If we're using soft floats, then these operations shoudl be
+    softened during type legalization. They'll never get to
+    LegalizeVectorOps or LegalizeDAG so they don't need to be
+    Expanded there.
+
+Local changes in those files are marked with
+
+  // XXX BINARYEN
+
```

Comment:
I feel pretty uncomfortable about checking in large parts of llvm like this.

At the very least I think we would want some kind of script that can keep them in sync.   However, a much cleaner way to do this would be to simply depend on libLLVM.

Reply:
Yeah, I don't think this is a long-term plan. What the long-term plan is, is still unclear, I think. One option is to link against libllvm, however, as I make progress here I am starting to think that is less likely. To get DWARF emitting capabilities from libllvm we'd need to define a proper LLVM MC layer, and we probably don't want to do all that.

If we don't do that, the options would be: write our own DWARF code, use libdwarf or another library, or keep using bits and pieces of LLVM + our own code.

Reply:
Is that because libLLVM doesn't have the obj2yaml code in it?

Reply:
<facepalm> .. right because obj2yaml was written as debugging and testing tool only... and we are abusing that now I guess...  but I get that it was the tool that made sense here.

Reply:
llvm's obj2yaml code is also not good enough - I had to add a few lines to it. So a stock LLVM isn't enough anyhow. Yeah, overall it seems like this isn't how LLVM was meant to be used...

Reply:
I doubt there would be any objection to adding that code upstream. Another option would be to link against libLLVM but copy/fork the required obj2yaml code and keep just that in binaryen.